app/validators/questionnaire_validator.py

```diff
@@ -107,6 +108,43 @@ def validate_smart_quotes(self):
                         pointer=translatable_item.pointer,
                     )
 
+    def validate_html(self):
+        html_strings = []
+        schema_object = SurveySchema(self.schema_element)
+
+        # pylint: disable=invalid-string-quote
+        html_regex = re.compile(r"<[^>]*>")
+
+        for translatable_item in schema_object.translatable_items:
+            schema_text = translatable_item.value
+
+            values_to_check = [schema_text]
+
+            if isinstance(schema_text, dict):
+                values_to_check = schema_text.values()
+
+            for schema_text in values_to_check:
+                if schema_text and html_regex.search(schema_text):
+                    html_strings.append(
+                        {"pointer": translatable_item.pointer, "text": schema_text}
+                    )
+
+        if html_strings:
+            self.check_invalid_html_tags(html_strings)
+
+    def check_invalid_html_tags(self, html_strings):
+        strong = re.compile("<strong[^>]>*>|</strong>")
+        anchor = re.compile("<a[^>]>*>|</a>")
+        for html_string in html_strings:
+            if not strong.search(html_string["text"]) and not anchor.search(
```

Comment:
At the moment some other html tags can pass this validation if they are in the same string as `a` and `strong` so maybe we should check how many of these we have combined compared to total amount of html tags present and raise an error if they are not the same?
```python
    all_tags = re.compile("<([a-z0-9]+)(?=[\s>])(?:[^>=]|='[^']*'|=\"[^\"]*\"|=[^'\"\s]*)*\s?\/?>")

    if len(strong.findall(html)) + len(anchor.findall(html)) != len(all_tags.findall(html)):
```

Reply:
Updated and added some additional cases and tests with the schema 👍